<launch>
  <!-- === PARAMETERS === -->
  <!-- Change these if your bag name or camera frame differs -->
  <!-- <arg name="bag_file" default="$(find your_package_name)/bags/your_data.bag"/>
  <arg name="fixed_frame" default="cam_L_color_optical_frame"/> -->

  <!-- === 1. Nodelet Manager === -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>

  <!-- === 2. RGB-D to PointCloud2 === -->
  <node pkg="nodelet" type="nodelet" name="pointcloud_xyzrgb"
        args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager">
    <param name="approximate_sync" value="true"/>
    <remap from="image_rect_color" to="/cam_L/color/image_raw"/>
    <remap from="depth_registered" to="/cam_L/aligned_depth_to_color/image_raw"/>
    <remap from="camera_info" to="/cam_L/color/camera_info"/>
  </node>

  <!-- === 3. Rosbag Playback === -->
  <node pkg="rosbag" type="play" name="bag_player" args="--clock --loop $(arg bag_file)" />

  <!-- === 4. RViz Visualization === -->
  <!-- <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find your_package_name)/rviz/view_pointcloud.rviz">
    <env name="ROS_NAMESPACE" value="/"/>
  </node> -->
</launch>
